### PR TITLE
Adjustments

### PR DESCRIPTION
**Pull request by dpoetzsch**
First, thanks for the great work.

However, to make it apply to our project and workflow we needed more configuration options. This PR handles these:

1. An option to ignore empty values
2. An option to ignore the order of keys
3. An optional specification of the indentation used in json files
4. Detection of invalid tabs in values
5. Tests & documentation for all of these

I think in the long run an integration of something like [jsonlint](https://github.com/zaach/jsonlint) would be a good idea and also solve issue #27 

If this is interesting to you maybe I could work smth out with jsonlint in the next weeks (depending how much time I can spare)

### CODE REVIEW

**Comment by vincent-zurczak** [comment]
Hi! Thanks for this PR.
Overall, your upgrades seem correct to me. I will take a deeper look tomorrow and perform a code review.

File: lib/index.js [hunk 1]
```diff
@@ -449,6 +461,19 @@ function verify_i18n_one(options, content, filename) {
           'Values should be trimmed (no white space character at the beginning and the end). Key: ' + key, lineCpt);
     }
 
+    // Values must not contain tabs, see http://stackoverflow.com/a/19799355/3594403
+    if (value.match(/^\s*.+\t.+\s*$/)) {
+      failure = output_i18n_error(
+          options.cb,
+          'Values must not contain tabs (this is invalid json, escape them using \t). Key: ' + key, lineCpt);
```
Comment (vincent-zurczak) [inline lib/index.js]: "must not" => "cannot"

File: test/api.spec.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ describe('Validation on real projects', function() {
     assert.equal(result, false);
     assert.equal(errors.length, 2);
     assert.include(errors[0], 'i18n keys must all be in upper case (with only letters, numbers and underscores). Key: inv_key');
-    assert.include(errors[1], 'i18n keys must be sorted alphabetically. Key KEY_4 breaks this rule.');
+    assert.include(errors[1], 'i18n keys must be sorted alphabetically. Key inv_key breaks this rule.');
```
Comment (vincent-zurczak) [inline test/api.spec.js:113]: Why was this test updated?
The default behavior should not have changed.

File: test/api.spec.js [hunk 3]
```diff
@@ -456,4 +456,72 @@ describe('Validation on real projects', function() {
...
+    var options = {
+      loc_i18n: __dirname + '/resources/invalid_tabs',
+      loc_html: __dirname + '/resources/invalid_tabs',
+      cb: cb
+    };
+
+    var result = qual.validate(options);
+    assert.equal(result, false);
+    assert.equal(errors.length, 2);
+    assert.equal(errors[0], '2: Values must not contain tabs (this is invalid json, escape them using \t). Key: BAZ');
```
Comment (vincent-zurczak) [inline test/api.spec.js]: "must not" => "cannot"

File: readme.md [hunk 4]
```diff
@@ -45,6 +46,11 @@ This library does not support...
 * **options.check_html**: `true` to search non-translated text in HTML mark-ups and attributes,
 as well as in Angular texts (`{{ 'some text' }}`). Default is `true`. All the mark-ups are verified.
 About attributes, only **alt** and **title** are verified.
+* **options.indent**: The indentation used in translation files. Default is `\t`.
+* **ignore_order**: Normally keys should be arranged in alphabetical order to ease search for a specific key.
+Using this option this check can be deactivated. Default is `false`.
```
Comment (vincent-zurczak) [inline readme.md]: Just a detail, but add a comma after `Using this option`.

File: readme.md [hunk 5]
```diff
@@ -45,6 +46,11 @@ This library does not support...
...
+* **ignore_empty_values**: Normally values should not be empty. Using this option this check can be deactivated.
```
Comment (vincent-zurczak) [inline readme.md]: Just a detail, but add a comma after `Using this option`.

**Review by vincent-zurczak — CHANGES_REQUESTED** [review]

**Comment by vincent-zurczak** [comment]
OK. There are some little things. If you can fix them, I will merge your PR then. 👍 
If you have some time to work on json linting, your contribution will be welcome too.

File: test/api.spec.js (hunk 2 above)
Comment (dpoetzsch) [reply test/api.spec.js:113]: for me, it was broken from the beginning

**Comment by dpoetzsch** [comment]
I changed the mentioned things. Thanks for reviewing.

Still, I think "must not" is more apt as technically the values _can_ contain tabs, but they are not allowed to. However, your project, your rules :)

I will try to look into a jsonlint integration.

File: test/api.spec.js (hunk 2 above)
Comment (vincent-zurczak) [inline test/api.spec.js]: OK, I will check this later.

**Comment by vincent-zurczak** [comment]
So, maybe "should not" is better. I will fix it later.
The weird stuff is that tests are marked as successful while there is one failure. I created [an issue](https://github.com/linagora/angular-translate-quality/issues/29) about it.

See https://travis-ci.org/linagora/angular-translate-quality/builds/226516806
I guess this is the "inv_key" thing. Previous tests were all working: https://travis-ci.org/linagora/angular-translate-quality/builds/196362883

**Comment by dpoetzsch** [comment]
I can revert that "fix" of the test. However, for me locally it is broken then.

**Comment by vincent-zurczak** [comment]
Wait... I have taken a moment to investigate it. And you are right. The test was wrong.  :dizzy_face:
`i` comes before `k`. So, inv_key is the wrongly located key.

However, the previous test worked on my machine and on Travis. Looking at the code, I thought it might be an issue related to the localeCompare function. And indeed, there is a case-sensitive issue.

So, I am going to merge your PR.
I will also fix #29 and #30. And then, I will release a new version of the library on NPM.

Thank you for your help. :+1:

**Pull request merged**